src/validation.cpp

```diff
@@ -646,18 +646,20 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
             }
         }
 
-        // are the actual inputs available?
+        // This redundant check doesn't trigger the DoS code on purpose; if it did, it would make it easier
+        // for an attacker to attempt to split the network (Consensus::CheckTxInputs also checks this).
```

Comment:
I think you can change the `view.HaveInputs(tx)` below into a `Consensus::CheckTxInputs` call, doing some of the checks slightly earlier.